router/core/src/main/scala/com/twitter/finagle/buoyant/DstTracing.scala

```diff
@@ -21,9 +21,11 @@ object DstTracing {
       private[this] val localDtabShow = dst.localDtab.show
       private[this] val pathShow = dst.path.show
       override def apply(conn: ClientConnection) = {
-        Trace.recordBinary("namer.dtab.base", baseDtabShow)
-        Trace.recordBinary("namer.dtab.local", localDtabShow)
-        Trace.recordBinary("namer.path", pathShow)
+        if (Trace.isActivelyTracing) {
```

Comment:
`record` checks `isActivelyTracing` so we already shouldn't be annotating the request.  This does save us from constructing the annotation though. 

With this change. `debugTrace()` will no longer print annotations for non-sampled requests.